File: Projects/TransportathonHackathon.API/TransportathonHackathon.Domain/Entities/TransportRequest.cs
using Core.Persistence.Entities;

namespace TransportathonHackathon.Domain.Entities
{
    public class TransportRequest : Entity<Guid>
    {
        public Guid CustomerId { get; set; }
        public Guid CompanyId { get; set; }
        public Guid TransportTypeId { get; set; }
        public string CountryFrom { get; set; }
        public string CountryTo { get; set; }
        public string CityFrom { get; set; }
        public string CityTo { get; set; }
        public string PlaceSize { get; set; }
        public bool ApprovedByCompany { get; set; }
        public bool IsFinished { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? FinishDate { get; set; }

        public virtual Customer Customer { get; set; }
        public virtual Company Company { get; set; }
        public virtual TransportType TransportType { get; set; }
        public virtual Comment? Comment { get; set; }
        public virtual PaymentRequest? PaymentRequest { get; set; }
    }
}

File: Projects/TransportathonHackathon.Clients/angular-ui/src/app/models/domain-models/transportRequest.ts
import { Customer } from './customer';
import { Company } from './company';
import { TransportType } from './transportType';
import { Entity } from 'src/app/core/models/entity';
import { Comment } from './comment';

export interface TransportRequest extends Entity<string> {
  customerId: string;
  companyId: string;
  transportTypeId: string;
  countryFrom: string;
  countryTo: string;
  cityFrom: string;
  cityTo: string;
  placeSize: string;
  approvedByCompany: boolean;
  startDate: string;
  finishDate: string | null;
  customer: Customer;
  company: Company;
  transportType: TransportType;
  comment: Comment | null;
}
